tests/unit/core/mender-utils_test.cpp

```diff
@@ -68,3 +68,270 @@ TEST(MenderUtilsTest, WildcardComparison) {
+    ret = mender_utils_key_value_list_create_node("key2", "value2", &list);
+    EXPECT_EQ(ret, MENDER_OK);
+    ret = mender_utils_key_value_list_create_node("key1", "value1", &list);
+    EXPECT_EQ(ret, MENDER_OK);
+
+    char *key_value_str = nullptr;
+    ret                 = mender_utils_key_value_list_to_string(list, &key_value_str);
+    EXPECT_EQ(ret, MENDER_OK);
+    EXPECT_STREQ(key_value_str, "key1\x1Fvalue1\x1Ekey2\x1Fvalue2\x1Ekey3\x1Fvalue3\x1E");
+
+    free(key_value_str);
+    ret = mender_utils_key_value_list_free(list);
+    EXPECT_EQ(ret, MENDER_OK);
+}
+
+TEST(MenderUtilsTest, StringToKeyValueList) {
+
+    mender_err_t ret;
+
+    const char              *key_value_str = "key1\x1Fvalue1\x1Ekey2\x1Fvalue2\x1Ekey3\x1Fvalue3\x1E";
+    mender_key_value_list_t *list          = nullptr;
+    ret                                    = mender_utils_string_to_key_value_list(key_value_str, &list);
+    EXPECT_EQ(ret, MENDER_OK);
+
+    size_t       i_node = 3;
+    const size_t len    = sizeof(char *);
+    for (mender_key_value_list_t *item = list; item != nullptr; item = item->next) {
+
+        char key[len];
+        snprintf(key, len, "key%zu", i_node);
+        EXPECT_STREQ(item->key, key);
+
+        char value[len];
+        snprintf(value, len, "value%zu", i_node);
+        EXPECT_STREQ(item->value, value);
+
+        i_node--;
+    }
+
+    ret = mender_utils_key_value_list_free(list);
+    EXPECT_EQ(ret, MENDER_OK);
+}
+
+TEST(MenderUtilsTest, KeystoreFromJson) {
+    const char *json_str    = "{\"key1\": \"value1\", \"key2\": \"value2\"}";
+    cJSON      *json_object = cJSON_Parse(json_str);
+
+    mender_keystore_t *keystore = nullptr;
+    mender_err_t       ret      = mender_utils_keystore_from_json(&keystore, json_object);
+    EXPECT_EQ(ret, MENDER_OK);
+
+    EXPECT_EQ(mender_utils_keystore_length(keystore), 2);
+
+    EXPECT_STREQ(keystore[0].name, "key1");
+    EXPECT_STREQ(keystore[0].value, "value1");
+    EXPECT_STREQ(keystore[1].name, "key2");
+    EXPECT_STREQ(keystore[1].value, "value2");
+
+    cJSON_Delete(json_object);
+    mender_utils_keystore_delete(keystore);
+}
+
+TEST(MenderUtilsTest, KeystoreToJson) {
+    mender_keystore_t *keystore = mender_utils_keystore_new(2);
+    /* workaround for `ISO C++ forbids converting a string constant to ‘char*’` */
+    char key1[]   = "key1";
+    char value1[] = "value1";
+    mender_utils_keystore_set_item(keystore, 0, key1, value1);
+    char key2[]   = "key2";
+    char value2[] = "value2";
+    mender_utils_keystore_set_item(keystore, 1, key2, value2);
+
+    cJSON       *json_object = nullptr;
+    mender_err_t ret         = mender_utils_keystore_to_json(keystore, &json_object);
+
+    EXPECT_EQ(ret, MENDER_OK);
+    EXPECT_STREQ(cJSON_GetObjectItem(json_object, "key1")->valuestring, "value1");
+    EXPECT_STREQ(cJSON_GetObjectItem(json_object, "key2")->valuestring, "value2");
+
+    cJSON_Delete(json_object);
+    mender_utils_keystore_delete(keystore);
+}
```

Comment:
General comment,

I see that although we are using C++ for the tests, we are not making much use of it. At least I would think that `std::string` is quite handy for testing.

Reply:
Yup! Definitely nicer than `snprintf()` and friends. :)

Reply:
Right,  that makes sense, didn't think much of it when I wrote the tests. I "C++-ified" it now